### PR TITLE
Line material fix

### PR DESCRIPTION
**Pull request by zxx43**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

issue: https://github.com/cocos/3d-tasks/issues/13692

### CODE REVIEW

**Comment by SantyWang** [comment]
Line should inherite ModelRenderer

File: cocos/particle/line.ts [hunk 1]
```diff
@@ -119,152 +117,147 @@ export class Line extends Component {
...
+    public color = new GradientRange();
 
     @serializable
     private _tile = new Vec2(1, 1);
 
+    private _tile_offset: Vec4 = new Vec4();
+
     /**
      * @zh 图块数。
+     * @deprecated since v3.7
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Please describe the alternate in current version

File: cocos/particle/line.ts [hunk 2]
```diff
@@ -119,152 +117,147 @@ export class Line extends Component {
...
-            this._materialInstance.setProperty('mainTiling_Offset', this._tile_offset);
+            this.material.setProperty('mainTiling_Offset', this._tile_offset);
         }
     }
 
     @serializable
     private _offset = new Vec2(0, 0);
 
+    /**
+     * @deprecated since v3.7
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Please describe the alternate in current version

File: cocos/particle/line.ts [hunk 3]
```diff
@@ -22,62 +22,59 @@
...
+            }
         }
     }
-    @serializable
-    private _material: Material | null = null;
-    private _materialInstance: MaterialInstance | null = null;
 
+    @override
+    @visible(false)
     @type(Material)
```
Comment (SantyWang) [inline cocos/particle/line.ts]: seems `@type(Material)` not correct. It's a array. Should it be `@type([Material])`?

File: cocos/particle/line.ts [hunk 4]
```diff
@@ -119,152 +117,147 @@ export class Line extends Component {
...
      */
+    @serializable
     @type(CurveRange)
     @range([0, 1])
     @displayOrder(3)
     @tooltip('i18n:line.width')
-    get width () {
-        return this._width;
-    }
+    public width = new CurveRange();
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Why don't need setter?

File: cocos/particle/line.ts [hunk 5]
```diff
@@ -119,152 +117,147 @@ export class Line extends Component {
...
+        if (this._models.length > 0 && this._models[0]) {
+            const lineModel = this._models[0];
+            if (lineModel.scene) {
+                lineModel.scene.removeModel(lineModel);
+            }
+        }
+    }
+
+    protected update (dt: number): void {
+        if (this._models.length > 0 && this._models[0]) {
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Calcute the line vertices per frame is too wasted. Maybe update vertices when points changed.

File: cocos/particle/line.ts (hunk 3 above)
Comment (zxx43) [reply cocos/particle/line.ts]: @type(Material) is in parent class

File: cocos/particle/line.ts (hunk 2 above)
Comment (zxx43) [reply cocos/particle/line.ts]: modified

File: cocos/particle/line.ts (hunk 1 above)
Comment (zxx43) [reply cocos/particle/line.ts]: modified

File: cocos/particle/line.ts (hunk 3 above)
Comment (zxx43) [reply cocos/particle/line.ts]: removed type just use parent's

File: cocos/particle/line.ts (hunk 5 above)
Comment (zxx43) [reply cocos/particle/line.ts]: Changing value when use curve mode will not call the set function. Fix that in editor.

File: cocos/particle/line.ts [hunk 6]
```diff
@@ -140,131 +135,145 @@ export class Line extends Component {
...
+            const lineModel = this._models[0] as LineModel;
+            lineModel.addLineVertexData(this._positions, this._width, this._color);
+        }
+    }
+
+    @serializable
+    private _width = new CurveRange();
+
+    /**
+     * @zh 线段颜色。
```
Comment (SantyWang) [inline cocos/particle/line.ts:158]: Please add english version and more description

File: cocos/particle/line.ts [hunk 7]
```diff
@@ -140,131 +135,145 @@ export class Line extends Component {
...
+    private _color = new GradientRange();
+
     @serializable
     private _tile = new Vec2(1, 1);
 
+    private _tile_offset: Vec4 = new Vec4();
+
     /**
      * @zh 图块数。
+     * @deprecated since v3.7.2
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Please add english comment, and the reason why we deprecate it.

File: cocos/particle/line.ts [hunk 8]
```diff
@@ -140,131 +135,145 @@ export class Line extends Component {
...
-            this._materialInstance.setProperty('mainTiling_Offset', this._tile_offset);
+            this.material.setProperty('mainTiling_Offset', this._tile_offset);
         }
     }
 
     @serializable
     private _offset = new Vec2(0, 0);
 
+    /**
+     * @deprecated since v3.7.2
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Add the reason why we deprecate it.

File: cocos/particle/line.ts [hunk 9]
```diff
@@ -172,7 +175,8 @@ export class Line extends ModelRenderer {
 
     /**
      * @zh 图块数。
-     * @deprecated since v3.7.2
+     * @en Texture tile count.
+     * @deprecated since v3.7.2, duplicated with mainTiling_Offset in material.
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Since v3.7.2, please use the `mainTiling_Offset` option in material instead.

File: cocos/particle/line.ts [hunk 10]
```diff
@@ -195,7 +199,7 @@ export class Line extends ModelRenderer {
     private _offset = new Vec2(0, 0);
 
     /**
-     * @deprecated since v3.7.2
+     * @deprecated since v3.7.2, duplicated with mainTiling_Offset in material.
```
Comment (SantyWang) [inline cocos/particle/line.ts]: Since v3.7.2, please use the mainTiling_Offset option in material instead.

**Review by SantyWang — APPROVED** [review]

**Pull request merged**